egs/wsj/s5/local/wsj_format_local_lms.sh

```diff
@@ -42,16 +42,159 @@ if [[ -z $bos || -z $eos ]]; then
+    ngram-count -order 1 -text-has-weights -text $CLASSES_LM_DIR/counts.${CLASS} -lm $CLASSES_LM_DIR/1g.${CLASS} && gzip -f $CLASSES_LM_DIR/1g.${CLASS}
+
+    gunzip -c $CLASSES_LM_DIR/1g.${CLASS}.gz | \
+    arpa2fst - | fstminimize | fstrmepsilon | fstprint | \
+    local/wclass/convert_self-loop_to_two-state_fst.pl | \
+    local/wclass/replace_BOS_and_EOS_with_disambig_symbol.pl -bos_input_symbol "#${CLASS}" -bos_output_symbol "<eps>" -eos_input_symbol "#${CLASS}" -eos_output_symbol "<eps>" -remove_bos_weight -remove_eos_weight | \
+    fstcompile  -isymbols=${lang}/words.txt -osymbols=${lang}/words.txt --keep_osymbols=false > $CLASSES_LM_DIR/G.1g_${CLASS}.fst
+
+    # in case there is OOV modeling based on sub-word units (SWUs)
+    if [ $OOV_MODELING == 1 ]; then
+      # count occurences of the SWUs for the OOV model of the category
+      ngram-count -text $CLASSES_DIR/${CLASS}.swu -write $CLASSES_LM_DIR/counts.${CLASS}_SWU
+      ngram-count -order 1 -read $CLASSES_LM_DIR/counts.${CLASS}_SWU -lm $CLASSES_LM_DIR/1g.${CLASS}_SWU && gzip -f $CLASSES_LM_DIR/1g.${CLASS}_SWU
+
+      # create an FST from class-specific SWU sub-language model for OOV detection
+      gunzip -c $CLASSES_LM_DIR/1g.${CLASS}_SWU.gz | \
+      arpa2fst - | fstminimize | fstrmepsilon | fstprint | \
+      local/wclass/convert_self-loop_to_two-state_fst.pl -disambig_symbol "#${CLASS}_SWU_BACK" | \
+      local/wclass/replace_BOS_and_EOS_with_disambig_symbol.pl -bos_input_symbol "#${CLASS}_SWU" -bos_output_symbol "#${CLASS}_SWU" -eos_input_symbol "#${CLASS}_SWU" -eos_output_symbol "#${CLASS}_SWU" -remove_bos_weight -remove_eos_weight | \
+      fstcompile -isymbols=${lang}/words.txt -osymbols=${lang}/words.txt --keep_osymbols=false > $CLASSES_LM_DIR/G.1g_${CLASS}_SWU.fst
+
+      # the ${CLASS}_SWU label is an integer -> we need that number for 'fstreplace'
+      SWU_MODEL_LABEL=`awk -v unk="${CLASS}_SWU" '{ if ( $1 == unk ) { print $2 } }' ${lang}/words.txt`
+      echo "- Extracted label for ${CLASS}_SWU: \"${SWU_MODEL_LABEL}\""
+
+      if [ -n "${SWU_MODEL_LABEL}" ]; then
+        fstreplace --epsilon_on_replace $CLASSES_LM_DIR/G.1g_${CLASS}.fst 0 ${CLASSES_LM_DIR}/G.1g_${CLASS}_SWU.fst ${SWU_MODEL_LABEL} |\
+        fstrmepsilon |\
+        fstminimizeencoded > $CLASSES_LM_DIR/G.1g_${CLASS}_new.fst
+        mv $CLASSES_LM_DIR/G.1g_${CLASS}_new.fst $CLASSES_LM_DIR/G.1g_${CLASS}.fst
+      else
+        echo Could not extract label for ${CLASS}_SWU, exiting... && exit 1;
+      fi
+    fi
+  done
+
+  echo "After word class LM training"
+fi
+
+
 # Be careful: this time we dispense with the grep -v '<s> <s>' so this might
 # not work for LMs generated from all toolkits.
 gunzip -c $lm_srcdir_3g/lm_pr6.0.gz | \
   arpa2fst --disambig-symbol=#0 \
            --read-symbol-table=$lang/words.txt - data/lang${lang_suffix}_test_bd_tgpr/G.fst || exit 1;
+
+if [ $CLASSES == 1 ]; then
```

Comment:
There is a lot of inconsistency in capitalization.  We prefer shell variables to be lower case-- yours are quite inconsistent.  And shell and python scripts should be lower case and with underscores, not dashes.